psci/Main.hs

```diff
@@ -92,6 +129,267 @@ getCommand singleLineMode = handleInterrupt (return (Right Nothing)) $ do
     go :: [String] -> InputT m String
     go ls = maybe (return . unlines $ reverse ls) (go . (:ls)) =<< getInputLine "  "
 
+-- | Make a JavaScript bundle for the browser.
+bundle :: IO (Either Bundle.ErrorMessage String)
+bundle = runExceptT $ do
+  inputFiles <- liftIO (glob (".psci_modules" </> "node_modules" </> "*" </> "*.js"))
+  input <- for inputFiles $ \filename -> do
+    js <- liftIO (readUTF8File filename)
+    mid <- Bundle.guessModuleIdentifier filename
+    length js `seq` return (mid, js)
+  Bundle.bundle input [] Nothing "PSCI"
+
+-- TODO: use JMacro here?
+indexJS :: IsString string => Int -> string
+indexJS serverPort = fromString . unlines $
+  [ "var get = function get(uri, callback, onError) {"
+  , "  var request = new XMLHttpRequest();"
+  , "  request.addEventListener('load', function() {"
+  , "    callback(request.responseText);"
+  , "  });"
+  , "  request.addEventListener('error', onError);"
+  , "  request.open('GET', uri);"
+  , "  request.send();"
+  , "};"
+  , "var evaluate = function evaluate(js) {"
+  , "  var buffer = [];"
+  , "  console.log = function(s) {"
```

Comment:
Might it make sense to restore `console.log` to its original value after evaluating, so that if people try to do stuff inside the browser window directly, it does what they would expect?